src/zenml/models/v2/base/base.py

```diff
@@ -126,6 +124,102 @@ def get_hydrated_version(self) -> "BaseResponse[AnyBody, AnyMetadata]":
             "using/hydrating the model."
         )
 
+    def get_body(self) -> "AnyBody":
+        """Fetch the body of the entity.
+
+        Returns:
+            The body field of the response.
+
+        Raises:
+            RuntimeError: If the body was not included in the response.
+        """
+        if not self.body:
+            raise RuntimeError(
+                f"Missing response body for {type(self).__name__}."
+            )
+
+        return self.body
+
+    def get_metadata(self) -> "AnyMetadata":
```

Comment:
The `get_body` and `get_metadata` method in the BaseIdentifiedResponse class are mostly duplicates of those in the base class. Could you please just call these method from those methods instead ?

Second idea: what about keeping `permission_denied` in the base class, in which case you won't need to override these methods at all in the `BaseIdentifiedResponse` class ? Or don't you think we'll need permissions for in-memory models too ?

Reply:
sure, only differences are the error messages.

Reply:
done (the first part)